pkg/entities/message.go

```diff
@@ -33,21 +33,31 @@ type Message struct {
 // Does it need an interface?
 
 type MsgBuffer struct {
-	buffer      bytes.Buffer
-	dataRecFlag bool
+	buffer       bytes.Buffer
+	dataRecFlag  bool
+	numOfRecords uint32
```

Comment:
That's correct. I use numOfRecords here to make it easier to calculate sequence number when sending message. If we use numOfSets and numOfRecords, we will need a way to get Sets from message and then get number of records, which is a little complicated since the only use here is to calculate sequence number. What do you think?